src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateArr = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
```

Comment:
Consider adding validation for the `actions` array to ensure that each action has the expected properties (`type`, `extraData`, `keysToRemove`). This will help prevent runtime errors and ensure that the function handles inputs gracefully.